debatable community ports part II

wersal454 opened the pull request:
## What type of PR is this?
- [x] Bug
- [x] Change
- [x] Feature
- [x] Miscellaneous

<details><summary><i><b>
Sub-categories:
</b></i></summary>

- [ ] Template
- [ ] Map
- [x] Config
- [x] Function
- [ ] Localization

</details>

## What have you changed, and why?

**Information:**

Ported community PR's:

https://github.com/official-antistasi-community/A3-Antistasi/pull/3389

https://github.com/official-antistasi-community/A3-Antistasi/pull/3390

https://github.com/official-antistasi-community/A3-Antistasi/pull/3393

https://github.com/official-antistasi-community/A3-Antistasi/pull/3395

https://github.com/official-antistasi-community/A3-Antistasi/pull/3406

https://github.com/official-antistasi-community/A3-Antistasi/pull/3416

https://github.com/official-antistasi-community/A3-Antistasi/pull/3430

https://github.com/official-antistasi-community/A3-Antistasi/pull/3436

https://github.com/official-antistasi-community/A3-Antistasi/pull/3446

https://github.com/official-antistasi-community/A3-Antistasi/pull/3454

https://github.com/official-antistasi-community/A3-Antistasi/pull/3455

https://github.com/official-antistasi-community/A3-Antistasi/pull/3472

https://github.com/official-antistasi-community/A3-Antistasi/pull/3481

https://github.com/official-antistasi-community/A3-Antistasi/pull/3482

https://github.com/official-antistasi-community/A3-Antistasi/pull/3496

https://github.com/official-antistasi-community/A3-Antistasi/pull/3501

https://github.com/official-antistasi-community/A3-Antistasi/pull/3507

https://github.com/official-antistasi-community/A3-Antistasi/pull/3516

https://github.com/official-antistasi-community/A3-Antistasi/pull/3514

https://github.com/official-antistasi-community/A3-Antistasi/pull/3521

https://github.com/official-antistasi-community/A3-Antistasi/pull/3531

https://github.com/official-antistasi-community/A3-Antistasi/pull/3534

https://github.com/official-antistasi-community/A3-Antistasi/pull/3535

https://github.com/official-antistasi-community/A3-Antistasi/pull/3563

https://github.com/official-antistasi-community/A3-Antistasi/pull/3566

https://github.com/official-antistasi-community/A3-Antistasi/pull/3569

https://github.com/official-antistasi-community/A3-Antistasi/pull/3570

https://github.com/official-antistasi-community/A3-Antistasi/pull/3573

https://github.com/official-antistasi-community/A3-Antistasi/pull/3574

https://github.com/official-antistasi-community/A3-Antistasi/pull/3579

https://github.com/official-antistasi-community/A3-Antistasi/pull/3584

https://github.com/official-antistasi-community/A3-Antistasi/pull/3586

https://github.com/official-antistasi-community/A3-Antistasi/pull/3592

https://github.com/official-antistasi-community/A3-Antistasi/pull/3595

https://github.com/official-antistasi-community/A3-Antistasi/pull/3386

https://github.com/official-antistasi-community/A3-Antistasi/pull/3381

https://github.com/official-antistasi-community/A3-Antistasi/pull/3377

https://github.com/official-antistasi-community/A3-Antistasi/pull/3376

https://github.com/official-antistasi-community/A3-Antistasi/pull/855

https://github.com/official-antistasi-community/A3-Antistasi/pull/3502

https://github.com/official-antistasi-community/A3-Antistasi/pull/3571

https://github.com/official-antistasi-community/A3-Antistasi/pull/3589

**How can your changes be tested, or reproduced?**

> ...

**Does this PR include changes not authored by you?**

- [ ] No
- [x] Yes (See below)

<details><summary><i><b>
Further info:
</b></i></summary><br>

- [x] I confirm that I, and by extension this repository, can legally use these third-party changes. (Provide links or author attribution.)

> ...

</details>

## Please verify the following (If possible).

`*` - Mandatory

- [ ] These changes are my own, or I have written permission to use them. `*`
- [x] These changes are ready for review, or will be marked as a draft. `*`
- [ ] I have loaded the mission in LAN host.
- [ ] I have loaded the mission on a dedicated server.

Is further work needed?

- [x] Needs further testing.
- [x] Needs further changes.
- [x] Needs to be converted to a draft.

<hr><details><summary><i><b>
Notes:
</b></i></summary><br>

> not ported:
#3392
#3458
#3459
#3493
#3515 (do I need to look at it?)
#3523 (need to look closely at this one)
#3527 (need to look closely at this one)
#3580
#3596
#3599
#3511 (may take some interesting ideas from it)
</details>

wersal454 committed:
community port #3389

wersal454 committed:
community port #3390

wersal454 committed:
community port #3393

wersal454 committed:
community port #3395

wersal454 committed:
community port #3406

wersal454 committed:
community port #3416

wersal454 committed:
community port #3430

debatable

wersal454 committed:
community port #3436

wersal454 committed:
community port #3446

needs to be tested

wersal454 committed:
community port #3454

work is needed on this one

wersal454 committed:
community port #3455

wersal454 committed:
community partial port  #3472

probaly needs testing

wersal454 committed:
debatable community port #3481

wersal454 committed:
community port #3482

wersal454 committed:
community port #3496

+ fixes

wersal454 committed:
community port #3501

wersal454 committed:
community port #3507

wersal454 committed:
community port #3516

wersal454 committed:
community port #3514

wersal454 committed:
community port #3521

wersal454 committed:
community ehh port #3531

needs a additional work

wersal454 committed:
community port #3534

maybe needs additional work

wersal454 committed:
community port #3535

probably needs a little bit of work

wersal454 committed:
community port #3563

NEEDS ADDITIONAL WORK

wersal454 committed:
community port #3566

should probably need to delete weapon config fixies?

wersal454 committed:
partically ported #3569

wersal454 committed:
community port #3570

wersal454 committed:
community port #3573

need to triple check this one

wersal454 committed:
community port #3574

community version is a tit bit different

wersal454 committed:
community port #3579

wersal454 committed:
community port #3584

needs a sanity check

wersal454 committed:
community port #3586

wersal454 committed:
partially ported #3592

needs a sanity check
and a check overall

wersal454 committed:
community port #3595

wersal454 committed:
community port #3386

wersal454 committed:
community port #3381

I think we need this one ?

wersal454 committed:
#3377

wersal454 committed:
#3376

wersal454 commented:
this will probably include garage sell button as well

wersal454 committed:
emm

ok?

Tiny-DM commented:
Hello Wersal, I'm reaching out from the Official Antistasi Community.

I would like to politely request that each ported PR be cited in this PR's description. Treating all of this work as a lump-sum movement of data isn't really a great practice and disregards the original PR authors.

I saw that the New Battle Menu functions PR was mentioned in the notes. As the author of that PR and current maintainer of the New Battle Menu, I am explicitly denying permission for any section of that code (#3515 on community, my NBM functions PR, and to a lesser extent the rest of the UI) to be ported. Please reach out to me directly if you wish to speak further about this.

As for the garage sell button, under the APL-ND license applied to all parts of the garage code, this manner of including specific features is explicitly prohibited and I would not advise for spending effort trying to port it. Again, reach out directly if you wish to speak further or work something out.

The easiest way to reach me is through discord, I am `tiny.dm` there. Feel free to reach out if you have any further questions whatsoever. Thank you.

wersal454 commented:
> I would like to politely request that each ported PR be cited in this PR's description. Treating all of this work as a lump-sum movement of data isn't really a great practice and disregards the original PR authors.

commits contain links to community PRs, so I personally don't really see a point in doing so, but ok...

> I saw that the New Battle Menu functions PR was mentioned in the notes. As the author of that PR and current maintainer of the New Battle Menu, I am explicitly denying permission for any section of that code ([official-antistasi-community#3515](https://github.com/official-antistasi-community/A3-Antistasi/pull/3515) on community, my NBM functions PR, and to a lesser extent the rest of the UI) to be ported. Please reach out to me directly if you wish to speak further about this.

yeah, I would probably still won't be adding any stuff from 3515 be it too much busy work for the stuff we already do (kinda?) hence "do I need to look at it?"

> As for the garage sell button, under the APL-ND license applied to all parts of the garage code, this manner of including specific features is explicitly prohibited and I would not advise for spending effort trying to port it. Again, reach out directly if you wish to speak further or work something out.

I'm porting it over for code "parity" with community and because @SilenceIsFatto asked me to

Tiny-DM commented:
> commits contain links to community PRs, so I personally don't really see a point in doing so, but ok...

Thank you, I appreciate it.

> yeah, I would probably still won't be adding any stuff from 3515 be it too much busy work for the stuff we already do (kinda?) hence "do I need to look at it?"

Sure, whatever works. My message still stands.

> I'm porting it over for code "parity" with community and because @SilenceIsFatto asked me to

So here's the issue, it's a license. You just can't pick bits and pieces of it. It's written out quite explicitly in the terms of the license.
If you do want to copy it, your option is to copy over the entire garage, as-is from Community. The current version of all files marked as APL-ND, as found on the Community GitHub, can be redistributed, and only without modification.  So if you do want to, you are allowed to just copy the entire thing over as it exists right now in community. No picking features though. All or nothing.
So your two options there are to forgo the sell button addition entirely, continuing with the version you have now, or you can carbon copy everything from Community as it stands.

wersal454 commented:
> So here's the issue, it's a license. You just can't pick bits and pieces of it. It's written out quite explicitly in the terms of the license. If you do want to copy it, your option is to copy over the entire garage, as-is from Community. The current version of all files marked as APL-ND, as found on the Community GitHub, can be redistributed, and only without modification. So if you do want to, you are allowed to just copy the entire thing over as it exists right now in community. No picking features though. All or nothing. So your two options there are to forgo the sell button addition entirely, continuing with the version you have now, or you can carbon copy everything from Community as it stands.

To this day I wonder how @igorkis-scrts managed to get slightly altered garage version

Silence-112 commented:
> I would like to politely request that each ported PR be cited in this PR's description. Treating all of this work as a lump-sum movement of data isn't really a great practice and disregards the original PR authors.

@wersal454
I changed the template for exactly this reason, and quite frankly I'm annoyed you didn't take it seriously and renamed it to "The golf clubbing part" aswell as just ignoring the `> ...` part under the attribution checkbox. Please fill it in properly, now and in future.

> So here's the issue, it's a license. You just can't pick bits and pieces of it. It's written out quite explicitly in the terms of the license.
If you do want to copy it, your option is to copy over the entire garage, as-is from Community. The current version of all files marked as APL-ND, as found on the Community GitHub, can be redistributed, and only without modification. So if you do want to, you are allowed to just copy the entire thing over as it exists right now in community. No picking features though. All or nothing.
So your two options there are to forgo the sell button addition entirely, continuing with the version you have now, or you can carbon copy everything from Community as it stands.

This was exactly why I requested the sell button to be ported. As far as I'm aware, the only thing we're missing to be a 1-1 APL-ND "carbon copy" of the Community garage *is* the sell button. So with this button addition, we should be 1-1 with the Community garage in terms of the APL-ND files.

wersal454 commented:
Updated PR description to include mention of every community PR

wersal454 committed:
tiny tini part of #855

igorkis-scrts commented:
@wersal454 
I hadn't got any kind of altered or slightly altered version of Hakon's Garage as APL-ND strictly prohibites that ("no deriatives" part of the loicense).  As far as I can remember, I made changes only in MIT-licensed files ([fn_addVehicle.sqf](https://github.com/igorkis-scrts/A3-Antistasi-Plus/blob/unstable/A3A/addons/garage/Public/fn_addVehicle.sqf) and [config.inc](https://github.com/igorkis-scrts/A3-Antistasi-Plus/blob/unstable/A3A/addons/garage/Public/config.inc)) so Plus stuff could be integrated into garage. I vaguely remember that there were more code changes but it seems that with every case when some alterations of non-MIT code were needed, I made pull requests with them into A3A Community Version (for example - [https://github.com/official-antistasi-community/A3-Antistasi/pull/2453](https://github.com/official-antistasi-community/A3-Antistasi/pull/2453) to have them both in Community Version and Plus. ¯\_(ツ)_/¯
So, garage deriatives are only allowed to alter MIT-licensed files with other stuff just the same and intact as Community's A3A.

wersal454 commented:
I conducted a little investigation and turns out, I fell in false memory trap. In 2023 there were indeed basically no difference in garage code between plus and community (except MIT stuff). And when I started actually working on ultimate in early 2024, Tiny made a couple of changes in toggleconfirmbutton https://github.com/official-antistasi-community/A3-Antistasi/commit/d3ba88a89cd8e3fafd39a6dccd25e0505899c6e8#diff-dd53f2047258ce633c6c766d92111839d5403756e261ffc71a8f47e96410a400  and in onLoad (APL-ND)
so I guess, when I made a comparision between ultimate (at the time, basically Plus with sparkles) and community, it was already different enough

made a separate PR to port garage as a whole #614

wersal454 committed:
Merge branch 'unstable' into patch-84

wersal454 committed:
conflict merge fixes

wersal454 committed:
debug console v3

wersal454 committed:
fixes + partically ported #3502

Didn't touch any of the plane loadout and bombs configs

If @stutpip123 wants, he can look at those

wersal454 committed:
finish porting #3531

wersal454 committed:
community port #3571

wersal454 committed:
community port #3589

wersal454 committed:
kinda ported #3599 ?

untested

wersal454 committed:
#3508

stutpip123 commented:
a PR this big is impossible to properly review and nearly impossible to test. best to redo it and split each community PR change into its own PR here

wersal454 committed:
Merge branch 'unstable' into patch-84

wersal454 commented:
I don't think making PR for every single one is a good idea either. But splitting them in batches, 13-14 in each, is fine (they are mostly small ones). + PRs were already processed by the community

jwoodruff40 commented:
Closed for a more deliberate, explicit permission model moving forward. I'm engaging with community team for their feedback.

wersal454 commented:
> Closed for a more deliberate, explicit permission model moving forward. I'm engaging with community team for their feedback.

Personally (as creator of this PR) didn't receive any complains about this approach (besides one from @Tiny-DM above) which was already dealt with

but, oh well.
